pkg/controller/persistentvolume/framework_test.go

```diff
@@ -372,8 +375,8 @@ func checkEvents(t *testing.T, expectedEvents []string, ctrl *PersistentVolumeCo
 				glog.V(5).Infof("event recorder finished")
 				finished = true
```

Comment:
yes, when `finished` is `true` the loop exits. But `finished` is set at different condition now.

Before it was set when there was no event in channel. This has proven to be racy, I got the controller waiting for CPU to send an event and the event was missed in this loop. Now we wait 1 minute and only after that `finished` is set.

Of course, when sender closes the channel, `finished` is set immediately in both cases.